### DISCUSSION TITLE
Why LuaSnip can't parse regTrig correctly

### DISCUSSION THREAD

**Discussion by JuanZoran** [L3MON4D3/LuaSnip#814] [Q&A]
![Screenshot_20230309_162825](https://user-images.githubusercontent.com/107862700/223967584-b36e05ce-e5d5-42a1-b2a6-f5126ed1df4b.png)
![Screenshot_20230309_162838](https://user-images.githubusercontent.com/107862700/223967615-e1b8b36c-415c-499a-8760-410ab38a77d6.png)
![Screenshot_20230309_162842](https://user-images.githubusercontent.com/107862700/223967649-804e77ef-1c48-47e6-94bc-9a2bfc6b1594.png)
Why I try to test regTrig option for luasnip, but get the wrong result.

- version: neovim nightly
- system: Ubuntu 22.04
- terminal: wezterm

**Comment by leiserfg** [comment]
You have an extra ')'  if it's intentional you have to escape it like `\)`

**Reply by JuanZoran** [reply]
That's not a problem and I have noticed that.
I have tried to use the example in Luasnip `Example/snippet.slua`:
```lua
s({ trig = "c(%d+)", regTrig = true }, {
		t("will only expand for even numbers"),
	}, {
		condition = function(line_to_cursor, matched_trigger, captures)
			return tonumber(captures[1]) % 2 == 0
		end,
	}),
```
It doesn't work.

**Reply by leiserfg** [reply]
That is not what you have in the image you posted tho.

**Reply by JuanZoran** [reply]
even though I pass "%.(%w+))", It should be triggered by ".{string})" instead of "%...", right?

**Reply by leiserfg** [reply]
No, if you pass an extra `)` without escaping it's an invalid pattern for Lua so it won't match (actually it should  crash).

**Reply by JuanZoran** [reply]
Alright, that's my fault. But in the fact, it doesn't work in any case.

**Reply by leiserfg** [reply]
Please then clean your pattern and come back with a description of what you have and what you wanna achieve, otherwise, it's hard to help you.

**Reply by JuanZoran** [reply]
![Screenshot_20230309_173012](https://user-images.githubusercontent.com/107862700/223979589-7b1708a9-7dd7-4099-bfdc-4039e6e924fc.png)
![Screenshot_20230309_173022](https://user-images.githubusercontent.com/107862700/223979612-225e00e1-c618-417f-9f17-66baf55943ea.png)

I attempted to use the regTrig example from Example/snippets.lua in my code, but when I type the letter "c" followed by some numbers, the snippets that I expected do not appear. I would like to know the reason for this issue and possible solutions.

**Reply by leiserfg** [reply]
If you are talking about the completions then yes, that does not use the `regTrig` they are used only when checking the pattern right before expansion and not for listing in `cmp_luasnip` .
